src/services/jsonLinks.ts

```diff
@@ -6,24 +6,71 @@
 import { DocumentLink } from 'vscode-languageserver-types';
 import { TextDocument, ASTNode, PropertyASTNode, Range, Thenable } from '../jsonLanguageTypes';
 import { JSONDocument } from '../parser/jsonParser';
+import { IJSONSchemaService } from './jsonSchemaService';
+import { URI } from 'vscode-uri';
+import { existsSync as fileExistsSync } from 'fs';
+import { resolve as resolvePath } from 'path';
 
-export function findLinks(document: TextDocument, doc: JSONDocument): Thenable<DocumentLink[]> {
-	const links: DocumentLink[] = [];
+export class JSONLinks {
+	private schemaService: IJSONSchemaService;
+
+	constructor(schemaService: IJSONSchemaService) {
+		this.schemaService = schemaService;
+	}
+
+	public findLinks(document: TextDocument, doc: JSONDocument): Thenable<DocumentLink[]> {
+		return findLinks(document, doc, this.schemaService);
+	}
+}
+
+export function findLinks(document: TextDocument, doc: JSONDocument, schemaService?: IJSONSchemaService): Thenable<DocumentLink[]> {
+	const promises: Thenable<DocumentLink[]>[] = [];
+
+	const refLinks: DocumentLink[] = [];
 	doc.visit(node => {
-		if (node.type === "property" && node.keyNode.value === "$ref" && node.valueNode?.type === 'string') {
+		if (node.type === "property" && node.valueNode?.type === 'string' && node.keyNode.value === "$ref") {
 			const path = node.valueNode.value;
 			const targetNode = findTargetNode(doc, path);
 			if (targetNode) {
 				const targetPos = document.positionAt(targetNode.offset);
-				links.push({
+				refLinks.push({
 					target: `${document.uri}#${targetPos.line + 1},${targetPos.character + 1}`,
 					range: createRange(document, node.valueNode)
 				});
 			}
 		}
+		if (node.type === "property" && node.valueNode?.type === 'string' && schemaService) {
+			const pathNode = node.valueNode;
+			const pathLinks: DocumentLink[] = [];
+			const promise = schemaService.getSchemaForResource(document.uri, doc).then((schema) => {
+				if (schema) {
+					const matchingSchemas = doc.getMatchingSchemas(schema.schema, pathNode.offset);
+					matchingSchemas.forEach((s) => {
+						if (s.node === pathNode && !s.inverted && s.schema) {
+							if (s.schema.format === 'uri-reference') {
+								const path = resolvePath(pathNode.value);
```

Comment:
Just occurred to me that this should probably be relative to the document URI - not the current working directory. Will update.